app/session.server.ts

```diff
@@ -35,7 +40,101 @@ const { getSession, commitSession, destroySession } =
     },
   });
 
-export { getSession, commitSession, destroySession };
+const fiveMinutesInSeconds = 60 * 5;
+const fiveMinutesInMilliseconds = fiveMinutesInSeconds * 1000;
+
+const twoWeeksInSeconds = 60 * 60 * 24 * 14;
+const twoWeeksInMilliseconds = twoWeeksInSeconds * 1000;
+
+// Session last 2 weeks by default. For testing purposes, we can add a
+// DEBUG_SESSION environment variable to make the sessions last 5 minutes, which
+// is the minimum allowed time for a session
+const COOKIE_EXPIRATION = {
+  seconds: process.env.DEBUG_SESSION ? fiveMinutesInSeconds : twoWeeksInSeconds,
+  milliseconds: process.env.DEBUG_SESSION
+    ? fiveMinutesInMilliseconds
+    : twoWeeksInMilliseconds,
+};
+
+/**
+ * Use this to register a new user or log in an existing user. This creates a
+ * session cookie, which can then be used to authenticate further requests.
+ *
+ * This method has 2 potential side effects:
+ * - if the user doesn't exist, it creates a new User document in Firestore
+ * - if the user doesn't have a profile, it creates a new UserProfile document
+ *   in Firestore
+ */
+export async function createUserSession(idToken: string, accessToken: string) {
+  const token = await auth.createSessionCookie(idToken, {
+    expiresIn: COOKIE_EXPIRATION.milliseconds,
+  });
+  const session = await storage.getSession();
+  session.set("token", token);
```

Comment:
Here's one of the biggest changes: instead of storing the user's `idToken` on the session cookie, we create a server-side session with Firebase's SDK, and store the generated token on the session.

The rest of this method was moved from the `login.tsx` file to here for clarity.